### PR TITLE
feat(trends): Add event count per group backend

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

We supported event count per user in Trends, but didn't support the same type of analysis for groups. This made Group Anaytics second-class, which isn't right.

## Changes

Adds support for count per group to the backend.

## How did you test this code?

Added query tests.

### CODE REVIEW

File: ee/clickhouse/queries/column_optimizer.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def properties_used_in_filter(self) -> Counter[PropertyIdentifier]:
             # If groups are involved, they're also used
             #
             # See ee/clickhouse/queries/trends/util.py#process_math
-            if entity.math == "unique_group":
```
Comment (macobo) [inline ee/clickhouse/queries/column_optimizer.py:79]: Are you confident in this? Could we be sending dirty filters where math_group_type is sent but not actually queries?

File: posthog/queries/trends/util.py [hunk 2]
```diff
@@ -142,3 +142,10 @@ def ensure_value_is_json_serializable(value: Any) -> Optional[float]:
             capture_exception(exception)
         return None
     return value
+
+
+def determine_aggregator(entity: Entity, team: Team) -> str:
```
Comment (macobo) [inline posthog/queries/trends/util.py:147]: Nit: Could we make Team the first argument? Reasoning: it's the most 'generic' argument and a lot of functions accept teams - having it first creates consistency.

File: posthog/queries/person_query.py [hunk 3]
```diff
@@ -140,9 +140,8 @@ def is_used(self):
         if any(self._uses_person_id(prop) for prop in self._filter.property_groups.flat):
             return True
         for entity in self._filter.entities:
-            if entity.math in COUNT_PER_ACTOR_MATH_FUNCTIONS or any(
-                self._uses_person_id(prop) for prop in entity.property_groups.flat
-            ):
+            is_count_per_user = entity.math in COUNT_PER_ACTOR_MATH_FUNCTIONS and entity.math_group_type_index is None
```
Comment (macobo) [inline posthog/queries/person_query.py:143]: Same as in column_optimizer: Are we sure we clean the math_group_type_index in the frontend?

It feels like this is better fit via math, but not sure on the context on this.

**Review by thmsobrmlr — COMMENTED** [review]
As discussed: I could not verify this is working in an "end-to-end" manner. This might fully be me not being able to create the right query and up for you to look into once you're looking at the frontend side of this.

Here's what I did:
1. Added 4 event for 2 companies:
    ```js
    posthog.group('company', 'id:1000', { name: 'Company 1000' })
    posthog.capture('my_event')
    posthog.capture('my_event')
    posthog.capture('my_event')

    posthog.group('company', 'id:1001', { name: 'Company 1001' })
    posthog.capture('my_event')
    ```
2. Querying with [this query](http://localhost:8000/api/projects/1/insights/trend/?insight=TRENDS&interval=day&display=ActionsLineGraph&actions=%5B%5D&events=%5B%7B%22id%22%3A%22my_event%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22math%22%3A%22avg_count_per_actor%22%2C%20%22math_group_type_index%22%3A%204%7D%5D&filter_test_accounts=true), i.e.:
    ```json
    {
       "id": "my_event",
       "math": "avg_count_per_actor",
       "math_group_type_index": 4
    }
     ```
3. I expected an average of 2 events for today, but am getting back 4 events i.e. total count.

---

During review I found that some comments are outdated and updated them here: https://github.com/PostHog/posthog/pull/13125

File: posthog/queries/trends/util.py (hunk 2 above)
Comment (Twixes) [reply posthog/queries/trends/util.py:147]: I kind of agree, but I also wanted to maintain consistency with the ordering in `process_math()` and the call site – `_total_volume_query()` – so this order is intentional

File: ee/clickhouse/queries/column_optimizer.py (hunk 1 above)
Comment (Twixes) [reply ee/clickhouse/queries/column_optimizer.py:79]: Replied in the other comment.

File: posthog/queries/person_query.py (hunk 3 above)
Comment (Twixes) [reply posthog/queries/person_query.py:143]: Update: I reworked this to be simpler. Dropped reporting to Sentry for now.

**Review by macobo — APPROVED** [review]

**Comment by Twixes** [comment]
The failing test was broken by #13158. Let's fix that separately.

**Pull request merged**